### ISSUE TITLE
Crash after starting rpmx

### ISSUE THREAD

**Issue by dylanpiera** [griest024/PKMNEssentialsPlugin#6]
![image](https://cloud.githubusercontent.com/assets/2012848/11918493/095d9b72-a734-11e5-816e-c11cf2498c4d.png)

What am I doing wrong?

**Comment by griest024** [comment]
Ugh this repo is really messy. I'll look at it tomorrow and make sure that
you are using the right branch (master/develop)

**Comment by griest024** [comment]
Also it may be running with the wrong version of ruby. A lot of this code
is old. I'll look at it tmr

**Comment by griest024** [comment]
Are you running it with the .bat file or are you running start_rmxp.rb directly?

**Comment by dylanpiera** [comment]
Using the bat file

**Comment by griest024** [comment]
I can't reproduce the issue. Can you create a gist with your game.bat file and the start_rmxp.rb file?

**Comment by dylanpiera** [comment]
The gist: http://pastebin.com/eEy2vmFb (If i understood what a gist is correctly)
And i get the exact same one with start_rmxp.rb

**Comment by griest024** [comment]
Make sure lines 142-147 of data_importer_exporter.rb in plugins looks like [this](https://gist.github.com/griest024/66762b1d97eee628707b). What's happening is files is not being set to an array (which Dir.entries should return, even if directory is empty). Also make sure that the directory ~/Desktop/Pokemon_Game_Github/Ethernal-Madness/Data/ actually contains the .rxdata files like the maps and Game.

**Comment by griest024** [comment]
Also, can you check what version of ruby you are using?

**Comment by dylanpiera** [comment]
I'll check if the file is the same,
But using Ruby version: 2.2.3-p173-x64

**Comment by dylanpiera** [comment]
The lines weren't the same, I've edited them. But i believe i'm still getting the same error:
http://pastebin.com/sRHUN6qz

All data files should be in there. I haven't moved any of em. So if they're correct in pkmnessentials V15 that shoudn't be the problem.

Also if it wasn't clear, It does actually load the program on steam. the script crashes right after opening

**Comment by griest024** [comment]
Okay now create a new ruby script with this line and run it `p Dir.entries("directory of pkmnessentials data folder")`

**Comment by dylanpiera** [comment]
http://pastebin.com/7SmVbeRk

**Comment by griest024** [comment]
This is the line I would run on my machine: `p Dir.entries("C:/Users/Peter/Pokemon Virginia/Data")` Make sure you don't forget the quotes or /

**Comment by griest024** [comment]
It's also probably good to use absolute pathnames. Keep trying until you get ruby to at least find the data folder.

**Comment by dylanpiera** [comment]
Ah i made them \ \ \ instead / / /
http://pastebin.com/gwLPucEw

**Comment by griest024** [comment]
Okay that's good. Now change that block in data_importer_exporter.rb to [this](https://gist.github.com/griest024/66762b1d97eee628707b) and then the plugin. For testing purposes can you run start_rmxp.rb from the command line directly instead of using the bat file? There are some weird conversions between windows filenames and ruby's

**Comment by griest024** [comment]
Also make sure that `$PROJECT_DIR` is properly set on line 22 of start_rmxp.rb. Try using an absolute pathname if you are not already and ensure that in the ruby file your pathname has forward slashes /.

**Comment by griest024** [comment]
Actually I take that back. Because windows needs to use that pathname as well, use double backslash as a delimiter. So the path would be like `"C:\\Users\\me\\my pokemon game"`. This is only for $PROJECT_DIR though.

**Comment by dylanpiera** [comment]
The $PROJECT_DIR needs to be set to the plugins folder location or the actual location of the rpxp project file?

**Comment by griest024** [comment]
The location of the rmxp project, that is, the directory that holds the .rxproj file.

**Comment by griest024** [comment]
NOT the location of the plugin

**Comment by dylanpiera** [comment]
Well that was one mistake i had made, new error: http://pastebin.com/9DnQMe9d

**Comment by griest024** [comment]
You need to copy Game.yaml from the github project to the project folder. For reference [this](https://gist.github.com/griest024/c4ed02630bb70f191639) is what the file should look like.

**Comment by dylanpiera** [comment]
Now i get this: http://pastebin.com/M2jqWHBr

**Comment by griest024** [comment]
Only the backslashes \ should be double. All forward slashes / should be single

**Comment by dylanpiera** [comment]
Where?

**Comment by dylanpiera** [comment]
At the top? (Cause chancing that in Game.yalm doesn't help)

**Comment by griest024** [comment]
REDACTED\Pokemon_Game_Github\Ethernal-Madness//Data/
"REDACTED\Pokemon_Game_Github\Ethernal-Madness//Data/"
is what was in the pastebin. the 2 forward slashes between Ethernal-Madness and Data should only be one. Make sure you didn't add an extra slash at the end of $PROJECT_DIR. So it should be `$PROJECT_DIR = "folder\\folder\\project folder"` NOT `$PROJECT_DIR = "folder\\folder\\project folder/"`

**Comment by griest024** [comment]
Also it appears that Game.yaml still isn't being read properly because $DATA_IGNORE_LIST is nil. A quick workaround is to set `$DATA_INGORE_LIST = ["Scripts.rxdata","EditorScripts.rxdata","PkmnAnimations.rxdata","TilesetsTemp.rxdata","Game.rxdata"]` above that block of code in data_importer_exporter.rb [like so](https://gist.github.com/griest024/66762b1d97eee628707b)

**Comment by griest024** [comment]
Ensure that Game.yaml is in the same folder as your .rxproj file (the folder pointed to by $PROJECT_DIR)

**Comment by dylanpiera** [comment]
Added the `$DATA_INGORE_LIST`
http://pastebin.com/BgQjUSVU
i believe its still the same error tho

**Comment by griest024** [comment]
$DATA_IGNORE_LIST couldn't possibly still be nil. Did you set it before the `p $DATA_IGNORE_LIST` line? Did you spell everything correctly? If yes then just change `files -= $DATA_IGNORE_LIST` to `files -= ["Scripts.rxdata","EditorScripts.rxdata","PkmnAnimations.rxdata","TilesetsTemp.rxdata","Game.rxdata"]`

**Comment by dylanpiera** [comment]
No error, But it stops after launching rpxp. http://pastebin.com/dS5eD1Xf

**Comment by griest024** [comment]
Yes that is because you haven't modified any data files. To force export every time, comment out this line: `files = files.select { |e| file_modified_since?($INPUT_DIR + e, $STARTUP_TIME) }`

**Comment by dylanpiera** [comment]
So every time I use game.exe it'll save changes?

**Comment by griest024** [comment]
Only when you actually save the changes inside rpg maker

**Comment by griest024** [comment]
Just comment that line and it'll force export every time. You actually have to do this for the first time using it.

**Comment by dylanpiera** [comment]
So, For example

I open game.exe i edit an event in a random map i save it

( at this point nothing happens )
Then I restart game.exe and the src folder is updated?

**Comment by griest024** [comment]
This is the process of using the plugin. You run start_rmxp.rb (or the bat file), it imports yaml file in the src folder and converts it into .rxdata files into the Data folder. Then it opens rpgmaker xp. Then you make your changes, like editing a map or whatever, save, and close rpgmaker. Then the ruby script takes all the .rxdata files and converts them to yaml files and stores them in src/Data.

**Comment by dylanpiera** [comment]
"Then the ruby script takes all the .rxdata files and converts them to yaml files and stores them in src/Data."
The bat file shoudn't close here tho should it?

**Comment by griest024** [comment]
After all the files have been exported the window will probably close automatically, as the script has finished executing.

**Comment by dylanpiera** [comment]
Okay so thats supposed to happen

Let me try to actually test it with git.

**Comment by griest024** [comment]
Just verify that src/Data has a bunch of yaml files in it.

**Comment by dylanpiera** [comment]
It does

**Comment by griest024** [comment]
Then the plugin should be working now. I guess make a change to a map, like add a flower or something, and then verify that you can see that change in the commit after the files have been exported..

**Comment by dylanpiera** [comment]
It is not working. After changing something and saving rpxp, then re-opening game.exe still has the old version.

**Comment by griest024** [comment]
game.exe runs off the .rxdata files. I don't see how it could be affected by the plugin.

**Comment by dylanpiera** [comment]
Whenever i'm using game.exe it doesn't save

if i just run the rpxpproj file it does save

**Comment by griest024** [comment]
Make a repo of all the files in src. The intial commit is what's in there now. Then start the plugin and make a change to a map. Then save and after the files have been exported make a new commit to your src repo. You see that the appropriate yaml file has been changed.

**Comment by griest024** [comment]
How are you using game.exe to modify project files?

**Comment by griest024** [comment]
Note that Game.rxdata is actually ignored by the plugin because it contains custom classes that aren't in the official rgss api. I'm working on that.

**Comment by dylanpiera** [comment]
I'm moving an event on a map. Is that included in Game.rxdata cause if so that might explain it.

I've made a repo out of the src folder made a first comit, then started game.exe edited (^ what i said above) and saved. Github desktop says no changes were made.

**Comment by griest024** [comment]
That's probably because those changes are stored in game.rxdata. I don't know for sure. To test the functionality of the plugin, try making a map change.

**Comment by dylanpiera** [comment]
Created a new map and edited some tiles in it.

No changes where made according to github. (When re-opening using game.exe it is not saved but when opening the rpxp project file it has the new map)

**Comment by dylanpiera** [comment]
Want a full log of the game.exe?

**Comment by griest024** [comment]
So the plugin is not exporting the changed map?

**Comment by dylanpiera** [comment]
Exactly

**Comment by griest024** [comment]
Hmm you don't have rpg maker and game.exe open at the same time do you?

**Comment by dylanpiera** [comment]
Game.exe runs, in the middle opens rpxp then sometime later just stops no mather if the rpmaker is opened or closed

**Comment by griest024** [comment]
you mean game.exe not game.bat right?

**Comment by dylanpiera** [comment]
Oh. I've been using game.bat ...

**Comment by dylanpiera** [comment]
But there is no game.exe in the plugin, Just the actual game.exe to play the game

**Comment by griest024** [comment]
yeah okay, just for now, use start_rmxp.rb instead of game.bat to start the plugin. I've had weird things happen with game.bat. Also, game.bat should actually be in the project folder (with the .rxproj file)

**Comment by griest024** [comment]
I have to go now, I should be back later

**Comment by dylanpiera** [comment]
Using start_rmxp.rb does the same thing

Moving game.bat into the project folder also does the same thing

Okay. Thanks for all the help thus far
